deferred.go

```diff
@@ -24,10 +31,12 @@ func (d *Deferred) MarshalCBOR(w io.Writer) error {
 }
 
 func (d *Deferred) UnmarshalCBOR(br io.Reader) (err error) {
-	// Reuse any existing buffers.
-	reusedBuf := d.Raw[:0]
-	d.Raw = nil
-	buf := bytes.NewBuffer(reusedBuf)
+	buf := deferredBufferPool.Get().(*bytes.Buffer)
+
+	defer func() {
+		buf.Reset()
+		deferredBufferPool.Put(buf)
+	}()
```

Comment:
[`WriteMajorTypeHeaderBuf` taking `buf`](https://github.com/whyrusleeping/cbor-gen/pull/96/files#diff-fb5f9f17634692bfc3985a25f89a213f2bd1a1b19e3cb4152ce3eb93a9532f16R67) seem to know precisely how many bytes are getting written. Ideally, we should grow the buffer before writing considering that function is called in a loop.

But doing that needs a bit more refactoring which seems to be beyond the scope of this PR. Please feel free to address this in a separate PR.

Reply:
`bytes.Buffer` will expand by, IIRC, at least 512 bytes each time, so it shouldn't really matter.